Lib/test/support/__init__.py

```diff
@@ -2501,6 +2502,27 @@ def skip_unless_symlink(test):
     msg = "Requires functional symlink implementation"
     return test if ok else unittest.skip(msg)(test)
 
+_buggy_ucrt = None
+def skip_if_buggy_ucrt(test):
+    """
+    Skip decorator for tests that use buggy strptime/strftime
+
+    If the UCRT bugs are present time.localtime().tm_zone will be
+    an empty string, otherwise we assume the UCRT bugs are fixed
+
+    See bpo-37552 [Windows] strptime/strftime return invalid
+    results with UCRT version 17763.615
+    """
+    global _buggy_ucrt
```

Comment:
This is not a particularly expensive check, right? Why cache it instead of checking each time?

Reply:
The caching is simple and easy to follow; I'd rather keep it than assume it's not worthwhile :)